packaging/MacOs/build_macos.py

```diff
@@ -0,0 +1,119 @@
+    @param component_filename: Name of the pkg component package to bundle the app in
+    @param installer_filename: Name of the installer that contains the component package
+    """
+    pkg_build_executable = os.environ.get("PKG_BUILD_EXECUTABLE", "pkgbuild")
+    product_build_executable = os.environ.get("PRODUCT_BUILD_EXECUTABLE", "productbuild")
+    codesign_identity = os.environ.get("CODESIGN_IDENTITY")
+
+    # This builds the component package that contains UltiMaker-Cura.app. This component package will be bundled in a distribution package.
+    pkg_build_arguments = [
+        pkg_build_executable,
+        "--component",
+        f"{dist_path}/{app_filename}",
+        f"{dist_path}/{component_filename}",
+        "--sign", codesign_identity,
+        "--install-location", "/Applications",
+    ]
+    subprocess.run(pkg_build_arguments)
+
+    # This automatically generates a distribution.xml file that is used to build the installer.
+    # If you want to make any changes to how the installer functions, this file should be changed to do that.
+    # TODO: Use --product {property_list_file} to pull keys out of file for distribution.xml. This can be used to set min requirements
+    distribution_creation_arguments = [
+        product_build_executable,
+        "--synthesize",
+        "--package", f"{dist_path}/{component_filename}",  # Package that will be inside installer
+        f"{dist_path}/distribution.xml",  # Output location for sythesized distributions file
+    ]
+    subprocess.run(distribution_creation_arguments)
+
+    # This creates the distributable package (Installer)
+    installer_creation_arguments = [
+        product_build_executable,
+        "--distribution", f"{dist_path}/distribution.xml",
+        "--package-path", dist_path,  # Where to find the component packages mentioned in distribution.xml (UltiMaker-Cura.pkg)
+        "--sign", codesign_identity,
+        f"{dist_path}/{installer_filename}",
+    ]
+    subprocess.run(installer_creation_arguments)
+
+
+def code_sign(dist_path: str, filename: str) -> None:
+    """ Sign a file using apple codesign. This uses a different certificate to package signing."""
+    codesign_executable = os.environ.get("CODESIGN", "codesign")
+    codesign_identity = os.environ.get("CODESIGN_IDENTITY")
+
+    sign_arguments = [codesign_executable,
+                 "-s", codesign_identity,
+                 "--timestamp",
+                 "-i", filename,  # This is by default derived from Info.plist or the filename. The documentation does not specify which, so it is explicit here. **This must be unique in the package**
+                 f"{dist_path}/{filename}"]
+
+    subprocess.run(sign_arguments)
+
+
+def notarize_file(dist_path: str, filename: str) -> None:
+    """ Notarize a file. This takes 5+ minutes, there is indication that this step is successful."""
+    notarize_user = os.environ.get("MAC_NOTARIZE_USER")
```

Comment:
Make notarizing optional such that community members can also run it locally

Reply:
I defaulted NOTARIZE_INSTALLER to false so this won't happen by default